### DISCUSSION TITLE
Where does nocodb store project info\How to alter project info without using web gui?

### DISCUSSION THREAD

**Discussion by Amonlol** [nocodb/nocodb#4261] [Q&A]
Nocodb and postgreSQL running as docker containers on my cluster server. Nocodb uses postgreSQL DB, named 'noco' to store all of its' metadata. So let's say I've created new project, using external connection to a different postgreSQL DB named 'postgres'. Is it possible to somehow change project metadata without using\connecting to nocodb's web dashboard?

**Comment by o1lab** [comment]
You would have to change the NC_DB variable that you have used to point to `noco` pg database. However, it will act as a new instance. What is the reason for changing the meta database ?

**Reply by Amonlol** [reply]
Well, I've configured my database, added lookups, changed TEXT (from pg column) to noco types (Single select, Checkbox, etc.) and more. If postgres db server's ip address is changed (let's say because of different hosting), the moment I try to open the same project from restored db I get an exception, because credentials are the same but ip is now different. So I need to change credentials of existing project or copy metadata to a new project with new credentials.

The only thing I can edit from web is project name
![image](https://user-images.githubusercontent.com/72953500/200834150-7fd7dfc0-44f4-417f-9b6a-76c4b97a9d60.png)

TL;DR As I can tell, all projects' metadata are being stored as text\json in noco's db. I need to know which table and column I need to change to achive my goal.

**Reply by o1lab** [reply]
Credentials are stored in an encrypted way as you would like it to be secure. We are introducing project edit with credentials (we'd this earlier - but it was easy for users to mess around by changing the database. Hence we removed it). We are reintroducing this soon.

Why don't you re-create the project by connecting to external database. This would be easier.

**Reply by Amonlol** [reply]
> Why don't you re-create the project by connecting to external database. This would be easier.

Because this way I lose all my changes, which were configured from noco's web (lookups, noco column types, etc.), as I have said earlier. Metadata for new project with the same credentials wouldn't be the same, that's the reason I've asking this question. 

P.S. Just to confirm. Noco uses the same pg server which stores my project data, but I've divided noco metadata and my own data from each other. Noco uses db named 'noco' to store its' metadata, the other db is named 'postgres', which stores all my data, which is being presented to a new project via connection to external database.

**Reply by Amonlol** [reply]
Still waiting for an appropriate response. But now it looks like I need to reconfigure the whole noco's web view each time server ip migrates, which sucks.

**Comment by yachal-upson** [comment]
Just flagging this is something we feel likely we'd run into - seems super frustrating. 
(Changing database hosting location but wanting to maintain all project metadata related to it)
